pennylane/_grad.py

```diff
@@ -457,6 +457,12 @@ def finite_diff(f, N=1, argnum=0, idx=None, delta=0.01):
     second-order derivative of the callable function ``f`` using a centered finite
     difference approximation.
 
+    .. warning::
+
+        The ``qml.finite_diff()`` function is deprecated and will be removed in an
+        upcoming release. To compute *quantum* gradients using finite-differences
+        (that is, gradients of tapes or QNode), please see :func:`.gradients.finite_diff`.
```

Comment:
Realized the warning wasn't appearing in the docs (CC @Jaybsoni)

Reply:
Great catch! I completely missed this one. I must have been distracted or something, thanks for reviewing it @josh146